src/test/java/com/google/sps/tests/NewsTest.java

```diff
@@ -0,0 +1,145 @@
+  public final static String PART_HTML_PATH = "src/test/java/com/google/sps/data/partialNewsHTML.txt";
+  public final static String EMPTY_HTML_PATH = "src/test/java/com/google/sps/data/emptyHTML.txt";
+  public List<Article> someArticleList = new ArrayList<>(); 
+
+ /**
+  * Helper method that converts a file path of a text file to a String.
+  * @return String version of text in specified file.
+  */
+  private static String textToString(String filePath) throws IOException {
+    StringBuilder stringBuilder = new StringBuilder();
+    try (BufferedReader inputReader = new BufferedReader(new FileReader(filePath))) {
+      String inputLine;
+      while ((inputLine = inputReader.readLine()) != null) {
+        stringBuilder.append(inputLine);
+      }
+    } catch(IOException e) {
+      System.err.println(e);
+      return "";
+    }
+    return stringBuilder.toString();
+  }
+
+ /**
+  * Helper method to check the makeArticleList method in the NewsServlet class.
+  * Checks if makeArticleList on HTML text in the specified file has output equivalent to expectedList.
+  */
+  private static void checkMakeArticleList(String filePath, List<Article> expectedList) throws IOException {
+    NewsServlet ns = new NewsServlet();
+    String htmlString = textToString(filePath);
+    List<Article> articleList = ns.makeArticleList(htmlString);
+    // Check each class instance in each article object is equal.
+    Assert.assertEquals(articleList.size(), expectedList.size());
+    for(int i = 0; i < articleList.size(); i++) {
+      Article realArticle = articleList.get(i);
+      Article expectedArticle = expectedList.get(i);  
+      Assert.assertEquals(expectedArticle.getTitle(), realArticle.getTitle());
+      Assert.assertEquals(expectedArticle.getLink(), realArticle.getLink());
+    }  
+  }
+  
+ /**
+  * Set up for following tests.
+  * Populates someArticleList with Article entries.
+  */
+  @Before
+  public void setUp() {
+    someArticleList.add(new Article(
+      "The latest Instagram trend: putting your name on photos of frogs, cats and Harry Styles",
+      "news.google.com/articles/CBMib2h0dHBzOi8vd3d3LnRoZWd1YXJkaWFuLmNvbS91cy1uZXdzLzIwMjAvanVsLzIwL2luc3RhZ3JhbS13aGF0LWZyb2ctYXJlLXlvdS1pbnNpZGUtdGhlLWxhdGVzdC13YXktdG8td2FzdGUtdGltZdIBb2h0dHBzOi8vYW1wLnRoZWd1YXJkaWFuLmNvbS91cy1uZXdzLzIwMjAvanVsLzIwL2luc3RhZ3JhbS13aGF0LWZyb2ctYXJlLXlvdS1pbnNpZGUtdGhlLWxhdGVzdC13YXktdG8td2FzdGUtdGltZQ?hl=en-US&amp;gl=US&amp;ceid=US%3Aen"
+    ));
+    someArticleList.add(new Article(
+      "How a blue protein turns tree frogs bright green",
+      "news.google.com/articles/CBMiVmh0dHBzOi8vd3d3LnNjaWVuY2VtYWcub3JnL25ld3MvMjAyMC8wNy9ob3ctYmx1ZS1wcm90ZWluLXR1cm5zLXRyZWUtZnJvZ3MtYnJpZ2h0LWdyZWVu0gEA?hl=en-US&amp;gl=US&amp;ceid=US%3Aen"
+    ));
+  }
+
+ /**
+  * Check that when given no HTML text, makeArticleList returns an empty list.
+  */
+  @Test
+  public void articlesFromEmptyHTML() throws IOException {
+    NewsServlet ns = new NewsServlet();
```

Comment:
[Optional] You can consider using a method with the @BeforeEach tag to initialize a class-level NewsServlet variable. That way a new NewsServlet is created and initialized before every test method (marked by @Test).

Reply:
I did not know about this tag before, thanks!

Reply:
Upon further research, the @beforeeach tag is just @before for junit 5. In junit4, the @before tag runs once before each individual test, while the @beforeclass tag runs once before all the tests (i.e., if you had a class with ten tests, @before would run ten times and @beforeclass would run once).